src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp

```diff
@@ -16,18 +18,48 @@ namespace DB
  */
 bool ParserJSONPathMemberAccess::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 {
-    if (pos->type != TokenType::Dot)
+    // There's a special case, that a path member can begin with number
+    if (pos->type != TokenType::Dot && pos->type != TokenType::Number)
         return false;
+    if (pos->type != TokenType::Number)
+        ++pos;
 
-    ++pos;
+    ASTPtr member_name;
 
-    if (pos->type != TokenType::BareWord && pos->type !=TokenType::QuotedIdentifier)
-        return false;
+    if (pos->type == TokenType::Number)[[unlikely]]
+    {
+        for (const auto * c = pos->begin; c != pos->end; ++c)
+        {
+            if (*c == '.' && c == pos->begin)
+                continue;
+            if (!isNumericASCII(*c))
```

Comment:
as I understand here if number does not start with `.` then we have something like `$1abc`, which is not valid, but here it seems we allow such case? let's add a test and make sure it is not parsed

Reply:
I think `$` always follow by a dot or a square bracket

Reply:
> I think $ is always followed by a dot or a square bracket

this point is - it should be, but we need to make sure that if not - then this case returns a "cannot parse" error. So I suggested to add a test for this case.